pypdf/_page.py

```diff
@@ -2392,3 +2266,192 @@ def _get_fonts_walk(
         _get_fonts_walk(cast(DictionaryObject, obj[key]), fnt, emb)
 
     return fnt, emb  # return the sets for each page
+
+
+def handle_tj(
```

Comment:
Wouldn't it be preferable to have this function local to the page class or at least as an classmethod?
the code will be next to this calling function

Reply:
Why would you prefer that?

I just moved those parts to a separate `_text_extraction` submodule within this PR. I hope that I can separate the text extraction code from the other responsibilities of the page class at some point (not necessarily in this PR)